service/host.go

```diff
@@ -217,15 +230,25 @@ func (s *host) Start(waitForShutdown bool) (<-chan Exit, <-chan struct{}, error)
 	}()
```

Comment:
this is very specific to StartAsync() and has no impact on `Start` -> should also be moved out of this function